wagtail/admin/views/workflows.py

```diff
@@ -393,24 +399,86 @@ def delete_action(self):
         self.object.deactivate(user=self.request.user)
 
 
-def usage(request, pk):
-    workflow = get_object_or_404(Workflow, id=pk)
+class WorkflowPageFilterSet(PageIndexView.filterset_class):
+    content_type = PageTypeFilter()
 
-    editable_pages = page_permission_policy.instances_user_has_permission_for(
-        request.user, "change"
-    )
-    pages = workflow.all_pages() & editable_pages
-    paginator = Paginator(pages, per_page=10)
-    pages = paginator.get_page(request.GET.get("p"))
 
-    return render(
-        request,
-        "wagtailadmin/workflows/usage.html",
-        {
-            "workflow": workflow,
-            "used_by": pages,
-        },
-    )
+class WorkflowUsageView(PageIndexView):
+    permission_policy = page_permission_policy
```

Comment:
This is necessary if we want to reuse the `_annotate_queryset` logic from the page `IndexView` because it uses `self.permission_policy.explorable_instances`.